File: my-folder/3334-apple-redistribution-into-boxes/solution.java
class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {
        int apples = 0;
        for(int i : apple){
            apples += i;
        }

        Arrays.sort(capacity);
        int box = 0;
        for(int i = capacity.length - 1; i >= 0; i--){
            box++;
            apples -= capacity[i];
            if(apples <= 0) return box;
        }
        return box;
    }
}

File: my-folder/3334-apple-redistribution-into-boxes/solution.py
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        app = sum(apple)
        capacity = sorted(capacity, key=lambda x: -x)
        ans = 0
        
        for c in capacity:
            if app > 0:
                app -= c
                ans += 1
            else:
                break
        return ans
